bench/lib/bench.c

```diff
@@ -180,6 +187,47 @@ bench_mean(struct bench * B)
 	return (sum / B->num_seconds);
 }
 
+static int
+compar_uint64_t(const void * ap, const void * bp)
+{
+	const uint64_t a = *(const uint64_t *)ap;
+	const uint64_t b = *(const uint64_t *)bp;
+
+	if (a > b)
+		return (1);
+	else if (b > a)
+		return (-1);
+	else
+		return (0);
+}
+
+/**
+ * bench_median(B):
+ * Return the median number of ticks per second.
+ */
+uint64_t
+bench_median(struct bench * B)
+{
+	uint64_t median;
+	size_t midpos;
+
+	/* Make a copy of the ticks. */
+	memcpy(B->sorted, B->ticks, B->num_seconds * sizeof(uint64_t));
+
+	/* Sort the extra array. */
+	qsort(B->sorted, B->num_seconds, sizeof(uint64_t), compar_uint64_t);
+
+	/* Find the midpoint, or mean of nearest midpoints. */
+	midpos = B->num_seconds / 2;
+	if ((B->num_seconds % 2) == 0)
+		median = (B->sorted[midpos - 1] + B->sorted[midpos]) / 2;
```

Comment:
If `num_seconds` is `1`, I'm guessing we want to average `sorted[0]` and `sorted[1]`, not `sorted[-1]` and `sorted[0]`?

Reply:
fixed in a REBASE commit.